### PR TITLE
Change Softmax to LogSoftmax to use NLLLoss.

### LINKED ISSUE #3: Softmax or logsoftmax for auxiliary prediction
The current code uses softmax for auxiliary prediction and during training uses NLL Loss. But according to [here](http://pytorch.org/docs/master/nn.html) it seems that the right last layer to go with NLL is LogSoftmax instead?

### PR DESCRIPTION
**Pull request by daa233**
Fix #3.

From [here](http://pytorch.org/docs/master/nn.html#torch.nn.Softmax), it is `LogSoftmax` should be used for `NLLLoss`, instead of `Softmax`.

> This module doesn’t work directly with NLLLoss, which expects the Log to be computed between the Softmax and itself. Use LogSoftmax instead (it’s faster and has better numerical properties).

### CODE REVIEW

**Comment by chaeunl** [comment]
That's right. CrossEntropyLoss = LogSoftmax + NLLLoss.
